src/testing/tests.rs

```diff
@@ -2942,4 +2979,74 @@ pub mod generic_wallet_tests {
             .await
             .unwrap();
     }
+
+    #[async_std::test]
+    pub async fn test_big_amount<'a, T: SystemUnderTest<'a>>() {
+        let max_record = 2u64.pow(63) - 1;
```

Comment:
Is max record limited by something other than u64::MAX?  I think you mentioned this is the case, but just want to be sure what the reason we only can use 63 bits is